### DISCUSSION TITLE
Problems with 0.9.x version

### DISCUSSION THREAD

**Discussion by ahnpnl** [robisim74/qwik-speak#37] [Q&A]
### How to update new `locale` property in `speakContext`?

I see that in **0.9.0** the breaking change is the removal of `changeLocale`. Previously, this function allowsed to update `locale` in `speakContext`. Now we don't have any ways to update `locale` in `speakContext` anymore.

This can lead to the situation that users try to use `useSpeakContext` hook to access the value of `locale` but now the value is no longer updated after the last locale change.

### How to make translation texts to change when using SPA navigation with `useNavigate`?

When testing locally with SPA navigation with `useNavigate`, the `loadTranslation$` isn't executed which makes the new translation texts are not displayed on the page. 

I saw the doc only uses `location.reload` but that is full page reload and I think SPA navigation should be supported too.

**Comment by robisim74** [comment]
In Breaking changes it written: _To improve performance, the changeLocale function has been removed and it is no longer possible to change locale at runtime without navigating/reloading the page_ (SPA or MPA is indifferent). 

Not only was the `changeLocale `function removed, but also:
- `SpeakLocale `is no longer reactive
- `QwikSpeakProvider `no longer watches for locale change and no longer reloads files

Why this change? Because translation at `compile-time` is preferred over `runtime `translation:
- No subscriptions are created from the library
- Only the translation data needed at runtime is serialized
- With _inlining_, the chunks sent to the browser are created one for each language, making them much lighter

So we have better performance, and less memory usage.

Of course, I too have always thought that changing the language without reloading the page in SPA mode is more elegant, but:
- How many users change language? and if they change language I don't think they do it more than once
- Qwik is very performant: it has become difficult to notice the differences between re-rendering all the components of a page and reloading it (see the live example: https://qwik-speak.pages.dev/)

**Reply by ahnpnl** [reply]
ah I see, thanks for your clearer explanation:)
